plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeManager.java

```diff
@@ -93,7 +89,8 @@ public Exchange createExchange(ExchangeContext context, int outputPartitionCount
                 throw new TrinoException(GENERIC_INTERNAL_ERROR, "Failed to generate new secret key: " + e.getMessage(), e);
             }
         }
-        FileSystemExchange exchange = new FileSystemExchange(baseDirectory, exchangeStorage, context, outputPartitionCount, secretKey, executor);
+        Collections.shuffle(baseDirectories);
```

Comment:
This is not thread safe. Create a copy instead. Actually I would recommend creating a copy and shuffling inside the `FileSystemExchange` as it already creates a copy.

Reply:
Ah I forgot to update the PR.